pkg/vault/manager.go

```diff
@@ -154,7 +154,7 @@ func (m *DefaultManager) Save() error {
 
 func (m *DefaultManager) renewSecret(leaseID string) error {
 	secret, err := m.client.Sys().Renew(leaseID, int(m.lease.Seconds()))
-	if err != nil {
+	if err != nil || secret == nil {
 		log.Errorf("error renewing lease: %s", err)
 		fatalError := checkFatalError(err)
```

Comment:
do these functions cope with a nil error? or could that cause a panic?

Reply:
In testing, when using a nil error, it doesn't panic but we get ugly print out `error renewing token: %!s(<nil>)`

Reply:
Have set the error if it is nil